syntax/verilog_systemverilog.vim

```diff
@@ -138,71 +138,82 @@ endif
 syn keyword verilogObject      super
 syn match   verilogObject      "\<\w\+\ze\(::\|\.\)" contains=verilogNumber
 
-let s:verilog_function_task_dequalifier =
-    \  '\('
-    \ .    '\('
-    \ .        'extern\s\+\(\(pure\s\+\)\?virtual\s\+\)\?'
-    \ .        '\|'
-    \ .        'pure\s\+virtual\s\+'
-    \ .    '\)'
-    \ .    '\(\(static\|protected\|local\)\s\+\)\?'
-    \ .'\)'
-
-execute 'syn match verilogStatement "'.s:verilog_function_task_dequalifier.'\@<=\<\(task\|function\)\>"'
 
 syn match verilogStatement '\(typedef\s\+\)\@<=\<class\>'
+syn match verilogStatement 'interface\ze\s\+class\>'
+
+" Create syntax definition from g:verilog_syntax dictionary
+function! s:SyntaxCreate(name, verilog_syntax)
+    if exists('a:verilog_syntax[a:name]')
+        let verilog_syn_region_name = 'verilog'.substitute(a:name, '.*', '\u&', '')
+        for entry in a:verilog_syntax[a:name]
+            if exists('entry["match"]')
+                " syn-match definitions
+                let match = entry["match"]
+                let verilog_syn_match_cmd = 'syn match '.verilog_syn_region_name.' "'.match.'"'
+
+                if exists('entry["syn_argument"]')
+                    let verilog_syn_match_cmd .= ' '.entry["syn_argument"]
+                endif
+
+                execute verilog_syn_match_cmd
+            elseif exists('entry["match_start"]') && exists('entry["match_end"]')
+                " syn-region definitions
+
+                let region_start = entry["match_start"]
+                let region_end = entry["match_end"]
+
+                if verilog_systemverilog#VariableExists('verilog_quick_syntax')
+                    execute 'syn keyword verilogStatement '.region_start.' '.region_end
+                else
+                    let verilog_syn_region_cmd = 'syn region '.verilog_syn_region_name
+
+                    if exists('entry["highlight"]')
+                        let verilog_syn_region_cmd .= ' matchgroup='.entry["highlight"]
+                    endif
+
+                    let verilog_syn_region_cmd .=
+                        \  ' start="'.region_start.'"'
+                        \ .' end="'.region_end.'"'
+
+                    if exists('entry["syn_argument"]')
+                        let verilog_syn_region_cmd .= ' '.entry["syn_argument"]
+                    endif
+
+                    if (index(s:verilog_syntax_fold, a:name) >= 0 || index(s:verilog_syntax_fold, "all") >= 0)
+                        let verilog_syn_region_cmd .= ' fold'
+                    endif
+
+                    execute verilog_syn_region_cmd
+                endif
+            else
+                echoerr 'Incorrect syntax defintion for '.a:name
+            endif
+        endfor
+    end
+endfunction
 
 " Only enable folding if verilog_syntax_fold_lst is defined
 let s:verilog_syntax_fold=verilog_systemverilog#VariableGetValue("verilog_syntax_fold_lst")
```

Comment:
This script variable is still used below, in the complex syntax statements for block and conditional compiler directives. At the moment I am still considering these as exceptional cases, that will be kept out of the generic approach implemented in `SyntaxCreate`.